### PR TITLE
feat: adding possibility to read flat surface container

### PR DESCRIPTION
**Pull request by asalzburger**
This PR adds a function to the `JsonSurfacesReader` that allows to read in a flat container rather than a Hierarchy map:

```python
>>> import acts
>>> from acts import examples
>>> sOptions = acts.examples.SurfaceJsonOptions()
>>> sOptions.inputFile = "some_sensitive_mod.json"
>>> sOptions.jsonEntryPath = ["surfaces"]
>>> surfaces = acts.examples.rreadSurfaceVectorFromJson(sOptions)
>>> print(len(surfaces))
16668
>>> geoContext = acts.GeometryContext()
>>> viewConfig = acts.ViewConfig()
>>> acts.examples.writeSurfacesObj(surfaces, geoContext, viewConfig, "some-surfaces.obj")
```

will work with this out of the box.

### CODE REVIEW

File: Examples/Io/Json/include/ActsExamples/Io/Json/JsonSurfacesReader.hpp [hunk 1]
```diff
@@ -36,4 +36,12 @@ struct Options {
 Acts::GeometryHierarchyMap<std::shared_ptr<Acts::Surface>> read(
     const Options& options);
 
+/// @brief Read the sflat surfaces urfaces from the input file
+///
+/// @param inputFile is the input file to read from
+///
+/// @return  a vector of surfaces
+std::vector<std::shared_ptr<Acts::Surface>> readSurfaces(
```
Comment (AJPfleger) [inline Examples/Io/Json/include/ActsExamples/Io/Json/JsonSurfacesReader.hpp:44]: I think the function names could be a bit confusing:
`read()` - "Read the surfaces"
`readSurfaces()` - "Read the flat surfaces"

File: Examples/Io/Json/include/ActsExamples/Io/Json/JsonSurfacesReader.hpp [hunk 2]
```diff
@@ -36,4 +36,12 @@ struct Options {
 Acts::GeometryHierarchyMap<std::shared_ptr<Acts::Surface>> read(
     const Options& options);
 
+/// @brief Read the sflat surfaces urfaces from the input file
```
Comment (AJPfleger) [inline Examples/Io/Json/include/ActsExamples/Io/Json/JsonSurfacesReader.hpp:39]:
```suggestion
/// @brief Read the flat surfaces from the input file
```

File: Examples/Io/Json/include/ActsExamples/Io/Json/JsonSurfacesReader.hpp (hunk 1 above)
Comment (asalzburger) [reply Examples/Io/Json/include/ActsExamples/Io/Json/JsonSurfacesReader.hpp:44]: Good point, as I saw that the second method is not referenced either by run script in the repository, I can rename them to `readSurfaceHierarchyMap` and `readSurfaceContainer`, however, then I will probably introduce a `acts.examples.json` module.

Comment (asalzburger) [reply Examples/Io/Json/include/ActsExamples/Io/Json/JsonSurfacesReader.hpp]: done, submodule not yet included, this needs a general cleanup of the python modules.

**Review by AJPfleger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [32f67b3dc60e42c8a47d98d9b6b3718d41cb38f6]
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/8234624)!

|status|job|report|
|------|---|------|
|:red_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44154550)||
|:green_circle: | [report_pull_request](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153738)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153737)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153736)||
|:red_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153735)||
|:red_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153734)||
|:red_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153733)||
|:red_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153732)||
|:red_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153731)||
|:red_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153730)||
|:red_circle: | [test_ActsCheckObjectCountsCachedWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153729)||
|:red_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153728)||
|:red_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153727)||
|:red_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153726)||
|:red_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153725)||
|:red_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153724)||
|:red_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153723)||
|:red_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153722)||
|:red_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153721)||
|:red_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153720)||
|:red_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153719)||
|:red_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153718)||
|:red_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153717)||
|:red_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153716)||
|:red_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153715)||
|:red_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153714)||
|:red_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153713)||
|:red_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153712)||
|:red_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153711)||
|:red_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153710)||
|:red_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153709)||
|:red_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153708)||
|:red_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/44153707)||

**Pull request merged**